services/merak-agent/agent.go

```diff
@@ -38,21 +39,35 @@ import (
 
 var (
 	gRPCPort       = flag.Int("gRPC port", constants.AGENT_GRPC_SERVER_PORT, "The server port")
-	prometheusPort = flag.Int("Prometheus port", constants.AGENT_PROMETHEUS_PORT, "The server port")
+	prometheusPort = flag.Int("Prometheus port", constants.PROMETHEUS_PORT, "The server port")
 )
 
 func main() {
+	var err error
+	val, ok := os.LookupEnv(constants.LOG_LEVEL_ENV)
+	if !ok {
+		log.Println("No log level specified. Defaulting to INFO")
+		val = "INFO"
+	}
+	handler.MerakLogger, err = logger.NewLogger(logger.LevelEnvParser(val), "merak-agent")
+	if err != nil {
+		handler.MerakLogger.Fatal("Failed to create logger\n", "err", err)
+	}
 
-	_, ok := os.LookupEnv(constants.AGENT_MODE_ENV)
+	val, ok = os.LookupEnv(constants.MODE_ENV)
 	if !ok {
+		// Default to Alcor mode
+		val = constants.MODE_ALCOR
+	}
+	if val == constants.MODE_ALCOR {
 		startPlugin()
 	}
 
 	// Start gRPC Server
 	flag.Parse()
 	lis, err := net.Listen("tcp", fmt.Sprintf(":%d", *gRPCPort))
 	if err != nil {
-		log.Fatalln("ERROR: Failed to listen", err)
+		handler.MerakLogger.Fatal("ERROR: Failed to listen", err, err)
```

Comment:
@phudtran why you put two `err` objects in the parameters??

Reply:
Thanks for the catch